sdv/metadata/single_table.py

```diff
@@ -269,7 +269,8 @@ def set_primary_key(self, id):
             id (str, tuple):
                 Name (or tuple of names) of the primary key column(s).
         """
-        self._validate_key(id, 'primary')
+        if id is not None:
```

Comment:
I think it makes more sense to check if id is None inside the `_validate_key` method. That way you don't need any of the changes in lines 400-409.